glotaran/project/dataclasses.py

```diff
@@ -0,0 +1,125 @@
+"""Contains helper methods for dataclasses."""
+from __future__ import annotations
+
+import dataclasses
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Callable
+    from typing import TypeVar
+
+    DefaultType = TypeVar("DefaultType")
+
+
+def exclude_from_dict_field(
+    default: DefaultType = dataclasses.MISSING,  # type:ignore[assignment]
+) -> DefaultType:
+    """Create a dataclass field with which will be excluded from ``asdict``.
+
+    Parameters
+    ----------
+    default : DefaultType
+        The default value of the field.
+
+    Returns
+    -------
+    DefaultType
+        The created field.
+    """
+    return dataclasses.field(default=default, metadata={"exclude_from_dict": True})
+
+
+def file_representation_field(
```

Comment:
Consider renaming `_dataclass_file_representation_field` or something that more accurately describes its behavior.